orderbook/usecase/orderbook_usecase_test.go

```diff
@@ -401,42 +457,144 @@ func (s *OrderbookUsecaseTestSuite) TestGetActiveOrders() {
-		dec, err := osmomath.NewDecFromStr(str)
-		s.Require().NoError(err)
-		return dec
+	testCases := []struct {
+		name                 string
+		setupMocks           func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, client *mocks.OrderbookGRPCClientMock, tokensusecase *mocks.TokensUsecaseMock)
+		expectedError        error
+		expectedOrders       []orderbookdomain.LimitOrder
+		expectedIsBestEffort bool
+	}{
+		{
+			name: "failed to get active orders",
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, client *mocks.OrderbookGRPCClientMock, tokensusecase *mocks.TokensUsecaseMock) {
+				client.GetActiveOrdersCb = s.getActiveOrdersFunc(nil, 0, assert.AnError)
+			},
+			expectedError: &types.FailedToGetActiveOrdersError{},
+		},
+		{
+			name: "no active orders to process",
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, client *mocks.OrderbookGRPCClientMock, tokensusecase *mocks.TokensUsecaseMock) {
+				client.GetActiveOrdersCb = s.getActiveOrdersFunc(nil, 0, nil)
+			},
+			expectedError:        nil,
+			expectedOrders:       nil,
+			expectedIsBestEffort: false,
+		},
+		{
+			name: "failed to get quote token metadata",
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, client *mocks.OrderbookGRPCClientMock, tokensusecase *mocks.TokensUsecaseMock) {
+				client.GetActiveOrdersCb = s.getActiveOrdersFunc(orderbookdomain.Orders{s.newOrder()}, 1, nil)
+				withGetMetadataByChainDenom(tokensusecase, "quoteToken")
+			},
+			expectedError: &types.FailedToGetMetadataError{},
+		},
+		{
+			name: "failed to get base token metadata",
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, client *mocks.OrderbookGRPCClientMock, tokensusecase *mocks.TokensUsecaseMock) {
+				client.GetActiveOrdersCb = s.getActiveOrdersFunc(orderbookdomain.Orders{s.newOrder()}, 1, nil)
+				withGetMetadataByChainDenom(tokensusecase, "quoteToken")
+			},
+			expectedError: &types.FailedToGetMetadataError{},
+		},
+		{
+			name: "error on creating formatted limit order ( no error - best effort )",
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, client *mocks.OrderbookGRPCClientMock, tokensusecase *mocks.TokensUsecaseMock) {
+				client.GetActiveOrdersCb = s.getActiveOrdersFunc(orderbookdomain.Orders{s.newOrder()}, 1, nil)
+				withGetMetadataByChainDenom(tokensusecase, "")
+				orderbookrepository.GetTickByIDFunc = s.getTickByIDFunc(orderbookdomain.OrderbookTick{}, false)
+			},
+			expectedOrders:       []orderbookdomain.LimitOrder{},
+			expectedError:        nil,
+			expectedIsBestEffort: true,
+		},
+		{
+			name: "successful processing of active orders",
```

Comment:
Would love to see 2 extra test cases:
- 2 successful orders processed
- 2 processed but one is best-effort

Reply:
For 2 orders created this [one](https://github.com/osmosis-labs/sqs/pull/499/files#diff-5fa55bd6410efa835405589edaffbb41ba95f810344564647d2d711e8bc15b67R749), and updated best [effort](https://github.com/osmosis-labs/sqs/pull/499/files#diff-5fa55bd6410efa835405589edaffbb41ba95f810344564647d2d711e8bc15b67R705).